modules/simple-payments/simple-payments.php

```diff
@@ -286,6 +289,67 @@ function allow_sync_post_meta( $post_meta ) {
 		) );
 	}
 
+	/**
+	 * Enable Simple payments custom meta values for access through the REST API.
+	 * Field’s value will be exposed on a .meta key in the endpoint response,
+	 * and WordPress will handle setting up the callbacks for reading and writing
+	 * to that meta key.
+	 *
+	 * @link https://developer.wordpress.org/rest-api/extending-the-rest-api/modifying-responses/
+	 * @link https://developer.wordpress.org/reference/functions/register_meta/
+	 */
+	public function register_meta_fields_in_rest_api() {
+			register_meta( 'post', 'spay_price', array(
```

Comment:
not required, but we can add:

```php
'object_subtype' => self::$post_type_product,
```

so we don't add these to all posts.

Reply:
Added! Good catch.

So in WordPress 4.9.8, `object_subtype` [was added](https://make.wordpress.org/core/2018/07/27/registering-metadata-in-4-9-8/) together with new shorthand helper:

```php
register_post_meta( 'jp_pay_product', 'spay_price', array(
    ...
) );
```

Is the same as:
```php
register_meta( 'post', 'spay_price', array(
    'object_subtype' => 'jp_pay_product',
    ...
) );
```

...with the difference that prior to WordPress 4.9.8 where `object_subtype` isn't supported yet, the latter will expose`spay_price` meta field in REST API endpoints for ALL post types, not just for `jp_pay_product` custom post type.

Should we therefore rather `register_post_meta` and not have functioning API for Simple Payments block prior to WP 4.9.8 at all, or expose these fields via all post types in older WP versions. Suppose our meta fields have unique enough names (and not "price" or "email") so that we wouldn't accidentally expose wrong meta fields from wrong posts.